Lesson3/Task1/Program.cs

```diff
@@ -0,0 +1,148 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Task1 {
+    public class Account {
+        public enum Type {
+            Credit,
+            Debit
+        }
+        private static int counter = 0;
+        private int _id;
+        private double _balance;
+        private Type _type;
+
+        public int id {
+            get => _id;
+        }
+        public double balance {
+            get => _balance;
+            set { _balance = value; }
+        }
+        public Type type {
+            get => _type;
+            set { _type = value; }
+        }
+        private void Set_ID() { _id = (++counter); }
+        public Account() {
+            Set_ID();
+            balance = 0;
+            type = Account.Type.Credit;
+        }
+        public Account(double _value) {
+            Set_ID();
+            balance = _value;
+            type = Account.Type.Credit;
+        }
+        public Account(Type _type) {
+            Set_ID();
+            balance = 0;
+            type = _type;
+        }
+        public Account(double _value, Type _type) {
+            Set_ID();
+            balance = _value;
+            type = _type;
+        }
+        public bool Deposit(double amount) {
+            if (balance + amount < 0) {
+                if (type == Type.Debit) return false;
+                if (amount < 0) balance += 0.1 * ((balance > 0) ? (balance + amount) : amount);  // Bank takes 10% for using credit ;)
+            }
+            balance += amount;
+            return true;
+        }
+        public bool Withdraw(double amount) { return Deposit(-amount); }
```

Comment:
Если нравится писать код в одну строку, используйте лямбда-синтаксис.
```suggestion
        public bool Withdraw(double amount) => Deposit(-amount); 
```